src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -1353,9 +1382,17 @@ private static void AddProjectBuildTask(ProjectInstance traversalProject, Projec
             task.SetParameter("BuildInParallel", "True");
 
             task.SetParameter("ToolsVersion", GetToolsVersionAttributeForDirectMSBuildTask());
-            task.SetParameter("Properties", GetPropertiesAttributeForDirectMSBuildTask(projectConfiguration));
 
-            if (outputItem != null)
+            if (projectConfiguration != null)
+            {
+                task.SetParameter("Properties", GetPropertiesAttributeForDirectMSBuildTask(projectConfiguration));
+            }
+            else
+            {
+                task.SetParameter("Properties", SolutionProperties);
```

Comment:
@rainersigwald suggested to add it.

> > This isn't actually needed as ProjectReference item already has those Properties as metadata.

> I'd really like to leave this in case a user injects a target that changes the properties, since it's been done that way for the normal build targets for a very long time.